tools/sphinx_ext/missing_references.py

```diff
@@ -37,7 +37,11 @@ def get_module_global_imports(module_import_path: str, reference_target_source_o
     """
     module = importlib.import_module(module_import_path)
     obj = getattr(module, reference_target_source_obj)
-    tree = _get_module_ast(inspect.getsourcefile(obj))
+
+    try:
+        tree = _get_module_ast(inspect.getsourcefile(obj))
+    except TypeError:
+        return set()
```

Comment:
If this function receives a `TypeVar` it `inspect.getsourcefile(obj)` raises `TypeError`.

IDK if this is OK @provinzkraut - I just added the handling to see if the docs would build/work with that in place, and they do. That is, `SecretT` _does_ document given this change, and not using stringized annotations with `bound=...`.

Reply:
Yeah, that's okay. This is a super janky way of fixing some of the Sphinx issue anyway so :woman_shrugging: